### PR TITLE
Cosmos DB: update 2019 08 preview

### PR DESCRIPTION
**Pull request by ChenTanyi**
https://github.com/Azure/azure-libraries-for-net/issues/896

### CODE REVIEW

File: src/ResourceManagement/CosmosDB/Domain/ICosmosDBAccount.cs [hunk 1]
```diff
@@ -105,6 +110,22 @@ public interface ICosmosDBAccount :
...
+        /// <return>The specific private endpoint connection.</return>
+        Task<Microsoft.Azure.Management.CosmosDB.Fluent.IPrivateEndpointConnection> GetPrivateEndpointConnectionAsync(string name, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <param name="groupName">Group name of private link resource.</param>
+        /// <return>The specific private link resource group.</return>
+        Microsoft.Azure.Management.CosmosDB.Fluent.IPrivateLinkResource GetPrivateLinkResource(string groupName);
+
+        /// <param name="groupName">Group name of private link resource.</param>
+        /// <return>The specific private link resource group.</return>
+        Task<Microsoft.Azure.Management.CosmosDB.Fluent.IPrivateLinkResource> GetPrivateLinkResourceAsync(string groupName, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (weidongxu-microsoft) [inline src/ResourceManagement/CosmosDB/Domain/ICosmosDBAccount.cs]: Do we have `<summary>` for these methods?

Same for below.

File: src/ResourceManagement/CosmosDB/Domain/IPrivateEndpointConnection.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        Microsoft.Azure.Management.ResourceManager.Fluent.Core.IHasInner<Microsoft.Azure.Management.CosmosDB.Fluent.PrivateEndpointConnectionInner>,
+        Microsoft.Azure.Management.ResourceManager.Fluent.Core.IExternalChildResource<Microsoft.Azure.Management.CosmosDB.Fluent.IPrivateEndpointConnection,Microsoft.Azure.Management.CosmosDB.Fluent.ICosmosDBAccount>
+    {
+
+        /// <summary>
+        /// Gets private endpoint which the connection belongs to.
+        /// </summary>
+        /// <summary>
+        /// Gets the privateEndpoint value.
+        /// </summary>
```
Comment (weidongxu-microsoft) [inline src/ResourceManagement/CosmosDB/Domain/IPrivateEndpointConnection.cs]: Eh, clean it up a bit?

File: src/ResourceManagement/CosmosDB/Domain/IPrivateLinkResource.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        Microsoft.Azure.Management.ResourceManager.Fluent.Core.IBeta,
+        Microsoft.Azure.Management.ResourceManager.Fluent.Core.IHasInner<Microsoft.Azure.Management.CosmosDB.Fluent.PrivateLinkResourceInner>
+    {
+
+        /// <summary>
+        /// Gets the private link resource group id.
+        /// </summary>
+        /// <summary>
+        /// Gets the groupId value.
+        /// </summary>
```
Comment (weidongxu-microsoft) [inline src/ResourceManagement/CosmosDB/Domain/IPrivateLinkResource.cs]: Clean up.

File: src/ResourceManagement/CosmosDB/Domain/PrivateEndpointConnection/Definition/IDefinition.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+namespace Microsoft.Azure.Management.CosmosDB.Fluent.PrivateEndpointConnection.Definition
+{
+    using Microsoft.Azure.Management.CosmosDB.Fluent;
+    using Microsoft.Azure.Management.CosmosDB.Fluent;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.ChildResource.Definition;
+    using Microsoft.Azure.Management.CosmosDB.Fluent;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.ChildResource.Definition;
+    using Microsoft.Azure.Management.CosmosDB.Fluent;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.ChildResource.Definition;
```
Comment (weidongxu-microsoft) [inline src/ResourceManagement/CosmosDB/Domain/PrivateEndpointConnection/Definition/IDefinition.cs]: Clean up.

File: src/ResourceManagement/CosmosDB/Domain/PrivateEndpointConnection/Update/IUpdate.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Management.CosmosDB.Fluent.PrivateEndpointConnection.Update
+{
+    using Microsoft.Azure.Management.CosmosDB.Fluent;
+    using Microsoft.Azure.Management.CosmosDB.Fluent;
+    using Microsoft.Azure.Management.CosmosDB.Fluent.CosmosDBAccount.Update;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.ChildResourceActions;
```
Comment (weidongxu-microsoft) [inline src/ResourceManagement/CosmosDB/Domain/PrivateEndpointConnection/Update/IUpdate.cs]: Clean up.

File: src/ResourceManagement/CosmosDB/Domain/PrivateEndpointConnection/UpdateDefinition/IUpdateDefinition.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Management.CosmosDB.Fluent.PrivateEndpointConnection.UpdateDefinition
+{
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.ChildResource.Update;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.ChildResource.Update;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.ChildResource.Update;
+    using Microsoft.Azure.Management.CosmosDB.Fluent;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.ChildResource.Update;
```
Comment (weidongxu-microsoft) [inline src/ResourceManagement/CosmosDB/Domain/PrivateEndpointConnection/UpdateDefinition/IUpdateDefinition.cs]: Clean up.

File: src/ResourceManagement/CosmosDB/Domain/ICosmosDBAccount.cs (hunk 1 above)
Comment (ChenTanyi) [reply src/ResourceManagement/CosmosDB/Domain/ICosmosDBAccount.cs]: these get and list function is summarized in `<return>`, thus no additional `<summary>`

File: src/ResourceManagement/CosmosDB/CosmosDBAccountImpl.cs [hunk 7]
```diff
@@ -13,6 +13,7 @@ namespace Microsoft.Azure.Management.CosmosDB.Fluent
     using System;
     using Microsoft.Azure.Management.CosmosDB.Fluent.Models;
```
Comment (yaohaizh) [inline src/ResourceManagement/CosmosDB/CosmosDBAccountImpl.cs]: Please keep the using list ordered

**Pull request merged**